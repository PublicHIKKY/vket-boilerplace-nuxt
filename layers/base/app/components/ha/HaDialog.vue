<template>
  <div
    class="ha-dialog"
    @click.self="handleCloseDialog"
  >
    <div class="dialog-window">
      <slot>no content.</slot>
    </div>
  </div>
</template>

<script setup lang="ts">
const emit = defineEmits<{
  (e: 'close'): void
}>()

const handleCloseDialog = () => emit('close')
</script>

<style lang="scss" scoped>
@use '#base/app/assets/styles/variables' as v;
@use '#base/app/assets/styles/mixins' as m;

.ha-dialog {
  align-items: center;
  background: v.$black-undercoat;
  display: flex;
  height: 100vh;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  width: 100vw;
  z-index: v.$zindex-dialog;

  > .dialog-window {
    background-color: v.$white;
    max-height: 100%;
    max-width: 100%;
    overflow: auto;
    padding: 10px;
    width: auto;
  }
}
</style>
