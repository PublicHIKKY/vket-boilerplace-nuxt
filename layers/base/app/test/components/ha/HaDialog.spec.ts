import { mount } from '@vue/test-utils'
import { describe, expect, test } from 'vitest'
import HaDialog from '#base/app/components/ha/HaDialog.vue'

test('ref component', () => {
  expect(HaDialog).toBeTruthy()
})

test('mount component', () => {
  const wrapper = mount(HaDialog, {})
  expect(wrapper.getCurrentComponent()).toBeTruthy()
  expect(wrapper.html()).toMatchSnapshot()
})

test('slot', () => {
  const wrapper = mount(HaDialog, {
    slots: {
      default: '<div>slot content.</div>',
    },
  })
  expect(wrapper.text()).toContain('slot content.')
})

describe('event', () => {
  it('click backdrop emits close-dialog event', async () => {
    const wrapper = mount(HaDialog)
    await wrapper.get('.ha-dialog').trigger('click')
    expect(wrapper.emitted('close')).toBeTruthy()
  })
})
