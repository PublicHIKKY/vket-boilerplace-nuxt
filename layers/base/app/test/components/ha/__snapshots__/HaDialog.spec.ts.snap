// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `
"<div data-v-8d2cdb80="" class="ha-dialog">
  <div data-v-8d2cdb80="" class="dialog-window">no content.</div>
</div>"
`;
